src/ui/controls/edit.cpp

```diff
@@ -2121,8 +2117,8 @@ void CEdit::SetMultiFont(bool bMulti)
 
     if (bMulti)
     {
-        m_format.reserve(m_maxChar+1);
-        for (int i = 0; i <= m_maxChar+1; i++)
+        m_format.reserve(GetMaxChar());
+        for (int i = 0; i <= static_cast<int>(GetMaxChar()); i++)
```

Comment:
Since default character limit is to be assumed infinite (well, as close to infinite as possible in 32-bit numbers), you can't just start filling up infinite amount of memory. It should be dynamic like m_text, and their lengths should be kept in sync (which is probably the most complicated part of code required to fully remove the limit).

Maybe it would be a bit easier if you changed type of m_format to `std::basic_string<Gfx::FontMetaChar>`?